Content.Server/GameObjects/Components/Sound/FootstepModifierComponent.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Log;
+using Robust.Shared.Utility;
+using YamlDotNet.RepresentationModel;
+using Content.Server.Interfaces;
+using Content.Shared.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using Content.Server.GameObjects.EntitySystems;
+using Robust.Server.GameObjects.EntitySystems;
+using Content.Shared.Audio;
+using Robust.Shared.Prototypes;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+using Robust.Shared.Audio;
+
+namespace Content.Server.GameObjects.Components.Sound
+{
+    /// <summary>
+    /// Changes footstep sound
+    /// </summary>
+    public class FootstepModifierComponent : Component, IOnMove
+    {
+        #pragma warning disable 649
+        [Dependency] private readonly IPrototypeManager _prototypeManager;
+        #pragma warning restore 649
+        /// <inheritdoc />
+        /// 
+        private Random _footstepRandom;
+
+        public override string Name => "FootstepModifier";
+
+        public string _soundCollectionName;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _soundCollectionName, "footstepSoundCollection", "");
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _footstepRandom = new Random();
```

Comment:
You should seed this with the current entity UID AND time stamp to avoid getting two entities made at the same time the same seed.

```suggestion
            _footstepRandom = new Random(Owner.Uid.GetHashCode() ^ DateTime.Now.GetHashCode());
```